dev/run-tests.py

```diff
@@ -121,7 +121,7 @@ def determine_modules_to_test(changed_modules):
     if modules.root in modules_to_test:
         return [modules.root]
     return toposort_flatten(
-        {m: set(m.dependencies).intersection(modules_to_test) for m in modules_to_test}, sort=True)
+        dict((m, set(m.dependencies).intersection(modules_to_test)) for m in modules_to_test))
```

Comment:
I see you added these changes to fix the tests, but they're unrelated to the patches you're backporting.

They should, at the very least, be a separate PR, if they're really needed.